### PR TITLE
Introduce Github actions for AWS ECS CI/CD

### LINKED ISSUE #330: AWS ECS CI/CD setup

### PR DESCRIPTION
**Pull request by doogyb**
closes #330

### CODE REVIEW

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#
+# 4. Store an IAM user access key in GitHub Actions secrets named `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
+#    See the documentation for each action used below for the recommended IAM policies for this IAM user,
+#    and best practices on handling the access key credentials.
+
+name: Deploy to Amazon ECS Dev service
+
+on:
+  push:
+    branches: [ dev, 330_aws_ecs_ci_cd ]
```
Comment (mindrones) [inline .github/workflows/dev.yml]: I guess we need to remove `330_aws_ecs_ci_cd` :)

File: .github/workflows/dev.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    # paths: [ 'be/**', 'shared/**' ]
+
```
Comment (mindrones) [inline .github/workflows/dev.yml]: I think we need to re-enable this line?

File: be/.aws/hpmt_be_dev_task.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
+{
```
Comment (mindrones) [inline be/.aws/hpmt_be_dev_task.json:1]: If supported I'd use tabs in the jsons in `be/.aws`

File: be/package.json [hunk 4]
```diff
@@ -2,10 +2,11 @@
...
 		"@fastify/cors": "^8.2.1",
-		"@svizzle/atlas": "^0.9.1",
-		"@svizzle/file": "^0.14.2",
-		"@svizzle/geo": "^0.9.2",
-		"@svizzle/utils": "^0.20.0",
+		"@opensearch-project/opensearch": "^2.3.1",
+		"@svizzle/atlas": "^0.9.0",
+		"@svizzle/file": "^0.14.1",
+		"@svizzle/geo": "^0.9.1",
+		"@svizzle/utils": "^0.19.0",
```
Comment (mindrones) [inline be/package.json]: I think this is a rebase error in that we upgraded Svizzle's versions in a previous commit

File: be/src/conf.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+export const esEnv = process.env.BE_ENV || 'dev'; // dev, staging, production
 const domains = {
 	dev: 'https://hpmt.es.dev.dap-tools.uk:9200',
 	staging: 'https://hpmt.es.staging.dap-tools.uk:9200',
-	production: 'https://hpmt.es.production.dap-tools.uk:9200'
+	production: 'https://hpmt.es.production.dap-tools.uk:443'
 };
 export const domain = domains[esEnv];
-export const index = 'hpmt_gold_interim_v7';
+export const index = 'hpmt_gold_interim_v7'
```
Comment (mindrones) [inline be/src/conf.js]: I'd reinstate the final `;` :)

File: be/src/routes.js [hunk 6]
```diff
@@ -54,5 +54,6 @@ export const routes = (fastify, options, done) => {
 	fastify.get('/terms1_histogram2', terms1Histogram2Schema, getTerms1Histogram2);
 	fastify.get('/terms1_stats2', terms1Stats2Schema, getTerms1Stats2);
 	fastify.get('/terms1_terms2', terms1Terms2Schema, getTerms1Terms2);
+	fastify.get('/test', (_, reply) => reply.send('Shared workspace fix'));
```
Comment (mindrones) [inline be/src/routes.js]: Do we need this line?

File: be/src/hooks.js [hunk 7]
```diff
@@ -17,6 +17,8 @@ export const onRequest = async (request, reply) => {
 		case '/certified':
 			field = 'installer_id_hash'
 			break
+		case '/test':
+			return;
```
Comment (mindrones) [inline be/src/hooks.js]: Do we need this?

**Review by mindrones — CHANGES_REQUESTED** [review]
Pass 1

**Review by mindrones — APPROVED** [review]
Awesome, GTM!

File: .github/workflows/production.yml [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+#    The format should follow the output of `aws ecs register-task-definition --generate-cli-skeleton`.
+#    Replace the value of the `ECS_TASK_DEFINITION` environment variable in the workflow below with the path to the JSON file.
+#    Replace the value of the `CONTAINER_NAME` environment variable in the workflow below with the name of the container
+#    in the `containerDefinitions` section of the task definition.
+#
+# 4. Store an IAM user access key in GitHub Actions secrets named `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
+#    See the documentation for each action used below for the recommended IAM policies for this IAM user,
+#    and best practices on handling the access key credentials.
+
+name: Deploy to Amazon ECS Dev service
```
Comment (saabi) [inline .github/workflows/production.yml]: Shouldn't this say `Production service`?

File: .github/workflows/staging.yml [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+#    The format should follow the output of `aws ecs register-task-definition --generate-cli-skeleton`.
+#    Replace the value of the `ECS_TASK_DEFINITION` environment variable in the workflow below with the path to the JSON file.
+#    Replace the value of the `CONTAINER_NAME` environment variable in the workflow below with the name of the container
+#    in the `containerDefinitions` section of the task definition.
+#
+# 4. Store an IAM user access key in GitHub Actions secrets named `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
+#    See the documentation for each action used below for the recommended IAM policies for this IAM user,
+#    and best practices on handling the access key credentials.
+
+name: Deploy to Amazon ECS Dev service
```
Comment (saabi) [inline .github/workflows/staging.yml]: Same as above: `Staging service`

**Review by saabi — APPROVED** [review]
Once the review comments are resolved, GTM

**Pull request merged**